### PR TITLE
ra-data-hasura: add custom primary key support

### PR DESCRIPTION
**Pull request by praveenweb**
### Description

This PR adds support to ra-data-hasura tool to accept a different primary key than `id`. There's a simple data transformation that converts other primary key field to `id` response, since react-admin enforces `id` to be mandatory in the response.

Usage:
```javascript
const headers = {'content-type': 'application/json'};
const config = { 'primaryKey': { 'tableName': 'primaryKeyColumn', 'tableName2': 'primaryKeyColumn' } };
const App = () => (
  <Admin
    dataProvider={hasuraDataProvider('http://localhost:8080', headers, config)}
    authProvider={authProvider}
    dashboard={Dashboard}
  >
```

If no custom primary key config is provided for tables, `id` will be considered as default.

### Affected components 

- Community Content

### CODE REVIEW

File: community/tools/ra-data-hasura/src/index.js [hunk 1]
```diff
@@ -32,6 +30,32 @@ export default (serverEndpoint, headers) => {
       throw new Error(JSON.stringify({'error': 'Invalid table/schema resource'}));
     }
 
+    return {schema: schema, tableName: tableName};
+
```
Comment (aam1r) [inline community/tools/ra-data-hasura/src/index.js]: Nitpicking here: You can remove this extra blank line here; it doesn't conform to the rest of the code's style

File: community/tools/ra-data-hasura/src/index.js [hunk 2]
```diff
@@ -32,6 +30,32 @@ export default (serverEndpoint, headers) => {
...
+  };
+
+  const getPrimaryKey = (resource) => {
+
```
Comment (aam1r) [inline community/tools/ra-data-hasura/src/index.js]: Nitpicking here: You can remove this extra blank line here; it doesn't conform to the rest of the code's style

File: community/tools/ra-data-hasura/src/index.js [hunk 3]
```diff
@@ -32,6 +30,32 @@ export default (serverEndpoint, headers) => {
...
+    let primaryKey = 'id';
```
Comment (aam1r) [inline community/tools/ra-data-hasura/src/index.js]: Suggestion: It'd be nice if you can define `id` as a constant somewhere at the top of this file.

Something like the follows above L10 would be excellent:

```javascript
const DEFAULT_PRIMARY_KEY = 'id';
```

This way, you can DRY up your code without having to re-write 'id' every time.

File: community/tools/ra-data-hasura/src/index.js [hunk 4]
```diff
@@ -32,6 +30,32 @@ export default (serverEndpoint, headers) => {
...
+
+    if (config && config['primaryKey'][resource]) {
```
Comment (aam1r) [inline community/tools/ra-data-hasura/src/index.js]: Nitpick:

```suggestion
    if (config && config.primaryKey[resource]) {
```

Again, this is a stylistic thing but considered a good practice. Refer to https://github.com/airbnb/javascript#properties--dot.

File: community/tools/ra-data-hasura/src/index.js [hunk 5]
```diff
@@ -32,6 +30,32 @@ export default (serverEndpoint, headers) => {
...
+      primaryKey = config['primaryKey'][resource];
+    }
+    return primaryKey;
```
Comment (aam1r) [inline community/tools/ra-data-hasura/src/index.js]: Nitpick: Blank line above L44

File: community/tools/ra-data-hasura/src/index.js [hunk 6]
```diff
@@ -32,6 +30,32 @@ export default (serverEndpoint, headers) => {
...
+  };
+
+  const convertDataRequestToHTTP = (type, resource, params) => {
+    const options = {};
+    let finalQuery = {};
+
+    const tableSchema = getTableSchema(resource);
+
+    let schema = tableSchema.schema;
```
Comment (aam1r) [inline community/tools/ra-data-hasura/src/index.js]: You can combine L53-L55 into a single using destructuring:

```javascript
let { schema, tableName } = tableSchema;
```

File: community/tools/ra-data-hasura/src/index.js [hunk 7]
```diff
@@ -42,8 +66,11 @@ export default (serverEndpoint, headers) => {
         finalSelectQuery.args.limit = params.pagination.perPage;
         finalSelectQuery.args.offset = (params.pagination.page * params.pagination.perPage) - params.pagination.perPage;
         finalSelectQuery.args.where = params.filter;
-        finalSelectQuery.args.order_by = {column: params.sort.field, type: params.sort.order.toLowerCase()};
+        // finalSelectQuery.args.order_by = {column: params.sort.field, type: params.sort.order.toLowerCase()};
```
Comment (aam1r) [inline community/tools/ra-data-hasura/src/index.js]: Any reason for leaving the old commented code in? This is generally considered a bad practice. If someone wants to refer to the previous version, they can look up the history using `git`.

References:

- https://stackoverflow.com/questions/3664288/is-it-bad-practice-to-leave-commented-out-code-in-production-releases
- https://www.nayuki.io/page/dont-share-commented-out-code
- https://kentcdodds.com/blog/please-dont-commit-commented-out-code

File: community/tools/ra-data-hasura/src/index.js [hunk 8]
```diff
@@ -32,6 +30,32 @@ export default (serverEndpoint, headers) => {
       throw new Error(JSON.stringify({'error': 'Invalid table/schema resource'}));
     }
 
+    return {schema: schema, tableName: tableName};
```
Comment (aam1r) [inline community/tools/ra-data-hasura/src/index.js]: You can trim this down to:

```suggestion
    return { schema, tableName };
```

This is due to https://alligator.io/js/object-property-shorthand-es6/ in ES6.

File: community/tools/ra-data-hasura/src/index.js (hunk 7 above)
Comment (praveenweb) [reply community/tools/ra-data-hasura/src/index.js]: Removed old comments. The PR was open for testing for @mnlbox to try it out :) So hadn't removed the old code temporarily.

File: community/tools/ra-data-hasura/src/index.js (hunk 6 above)
Comment (praveenweb) [reply community/tools/ra-data-hasura/src/index.js]: Makes sense 👍

**Pull request merged**